### PR TITLE
Cleanup and better password reset tokens

### PR DESCRIPTION
**Pull request by jackhorton**

### CODE REVIEW

File: src/controllers/auth.js [hunk 1]
```diff
@@ -155,129 +146,132 @@ export function logUserIn(attrs) {
...
+    ]).then((values) => {
+        if (!values[0]) {
+            throw new Error('Invalid reset token');
+        }
+
+        const expireDate = new Date(values[0].get('expire_date'));
+        const newPassword = values[1];
+
+        return values[0].destroy().then((resetToken) => {
+            if (new Date() > expireDate) {
```
Comment (garrickb) [inline src/controllers/auth.js:191]: Not sure if this is an issue, but the database time could be in a different time zone, so maybe get the current time with the knex command?

File: src/controllers/recipe.js [hunk 2]
```diff
@@ -1,76 +1,112 @@
...
+            }
+        });
     });
-});
+};
 
-router.get('/recipes/search/:searchTerm', (req, res) => {
-    request({uri: `${YUMMLY_RECIPE_SEARCH_URL}&q=${req.params.searchTerm}`,
-        timeout: 15000,
-        followRedirect: true,
```
Comment (garrickb) [inline src/controllers/recipe.js:39]: Yummly says that it's possible for there to be redirects, and we should be able to handle them properly.

File: src/controllers/recipe.js [hunk 3]
```diff
@@ -1,76 +1,112 @@
...
+
+        next(new Error('Unable to retrieve featured recipes'));
     });
 });
 
-router.get('/recipes/recipe/:recipeId', (req, res) => {
-    request({uri: `${YUMMLY_BASE_URL}recipe/${req.params.recipeId}?${YUMMLY_KEYS}&${YUMMLY_PARAMS}`,
-        timeout: 15000,
-        followRedirect: true,
-        maxRedirects: 10,
```
Comment (garrickb) [inline src/controllers/recipe.js:61]: Same thing about redirects here.

File: src/controllers/auth.js (hunk 1 above)
Comment (jackhorton) [inline src/controllers/auth.js:191]: The time zone is ISO, so im like ~95% sure it includes timezone parsing.

File: src/controllers/recipe.js (hunk 2 above)
Comment (jackhorton) [inline src/controllers/recipe.js:39]: I wrote the `yummly` function which does that automatically.

**Pull request merged**